### ISSUE TITLE
Tx/DialogState are backed up by array.Replace with enum

### ISSUE THREAD

**Issue by jaimecasero** [RestComm/jain-sip#119]
take care about serialization....

**Comment by jaimecasero** [comment]
@xhoaluu  would you like to work on this?

**Comment by xhoaluu** [comment]
@jaimecasero , I will work on that, this will help me to understand Tx before starting with issue #58

**Comment by jaimecasero** [comment]
Sure, go ahead. PRACK is advance feature on SIP. To become familiar with SIP, you may check SIP adventures articles which are quite illustrative. For transactions you may check :
[SIP Via](https://andrewjprokop.wordpress.com/2014/03/06/understanding-the-sip-via-header/)
[SIP Timers](https://andrewjprokop.wordpress.com/2013/07/02/understanding-sip-timers-part-i/)

Although not strictly speaking about TX state machine, they talk about things around the transaction. You will find more "adventures" linked there, so feel free to explore by yourself...

**Comment by xhoaluu** [comment]
@jaimecasero 

I also see that the TransactionState is backed up by array. Do I need to update it also?

**Comment by jaimecasero** [comment]
sure,thats why i wrote Tx/DialogState -> TransactionState and DialogState

**Comment by jaimecasero** [comment]
this is part of javax.sip API so lets keep this on hold by now
